### ISSUE TITLE
In {foreach $recList as $key=>$item} , $item cannot show all keys (ex: {$item.sid})

### ISSUE THREAD

**Issue by yourchoice** [smarty-php/smarty#149]
I have an array associated from php to smarty tpl. 

$recList have {$item.0|debug} : 

Array
(
    [feed_id] => 18124    
    [feed_ts] => 2014-04-16 10:02:59
    [site_id] => 18052
    [feed_truncate] => 0
    [lang_iso] => fr
    [feed_error] => XML KO
    [sid] => 18052
)

In {foreach} any keys can be showed, except {$item.sid}, that show null even is not null.

**Comment by artcs** [comment]
`{$foo.sid}` compiles to `<?php echo $_smarty_tpl->tpl_vars['foo']->value[sid];?>`
whereas `{$foo.bar}` compiles to `<?php echo $_smarty_tpl->tpl_vars['foo']->value['bar'];?>`
(Note the missing apostrophes around `sid`). Maybe it has to do something with the predefined SID from php sessions, which seems to be defined caseinsensitive.

Had this issue some time ago. You can work around it using`{$item['sid']}`, which compiles correct.

**Comment by yourchoice** [comment]
Yes, I saw that {$item['sid']} working.

In my case {$item.sid} come from an {foreach}.
>>
{foreach $recList as $key=>$item}
    #{$item.sid}#{$item['sid']}#<br>
{/foreach}
<<

I found a solution for my case,... idea is that these situations to be reviewed. Maybe exist and another predefined keys, even not understand how and why

**Comment by artcs** [comment]
Yes, this should be reviewed. But  the `{foreach}` part is not the culprit, it's a general bug when accessing tpl array vars with the dot syntax (`{$item.$key}`). 
`{$item[{$key}]}` will work inside `{foreach}` even if `$key` is `sid`.

**Comment by uwetews** [comment]
See https://github.com/smarty-php/smarty/issues/76

**Comment by artcs** [comment]
I would expect, to access constants one has to use `{$smarty.const.foo}`. The way it works now, one has to watch out not to define anything that has the same name as an array member. Is it possible to switch this behaviour off via `$php_handling` or some other switch? I stopped using `$array.foo` altogether because of this feature/bug.

Consider this example:
```php
define('foo', 'bar');
$foo = 'baz';
$Smarty->assign('foo', $foo);
```
Accessing `{$foo}` inside template gives you `baz` which is unexpected to me.

**Comment by artcs** [comment]
I would expect, to access constants one has to use `{$smarty.const.foo}`. The way it works now, one has to watch out not to define anything that has the same name as an array member. Is it possible to switch this behaviour off via `$php_handling` or some other switch? I stopped using `$array.foo` altogether because of this feature/bug.

Consider this example:
```php
define('foo', 'bar');
$foo = 'baz';
$arr = array('foo'=>'Foo','bar'=>'Bar', 'baz'=>'Baz');
$Smarty->assign('foo', $foo);
$Smarty->assign('arr', $arr);
```
Accessing `{$foo}` inside template correctly gives you `Baz` while {$arr.foo} results in `Bar` which is confusing to me as I cannot always know what some other developer might have defined somewhere in the code.

**Comment by uwetews** [comment]
I removed the inline constant support from the $foo.bar array syntax.

The fix is in the master branch an will later be included in 3.1.30

**Comment by artcs** [comment]
Perfect. Thank you very much.
